lib/src/edwards25519/field_element.dart

```diff
@@ -0,0 +1,1038 @@
+import 'package:solana/src/edwards25519/extensions.dart';
+
+const FieldElement sqrtM1 = FieldElement._fromConstList([
+  -32595792,
+  -7943725,
+  9377950,
+  3500415,
+  12389472,
+  -272473,
+  -25146209,
+  -2005654,
+  326686,
+  11406482,
+]);
+
+class SqrtRatioM1Result {
+  SqrtRatioM1Result(this.wasSquare, this.result);
+
+  final int wasSquare;
+  final FieldElement result;
+}
+
+class FieldElement {
+  FieldElement(this._t) {
+    if (_t.length < 10) {
+      throw Exception(
+        'cannot create a `FieldElement\' '
+        'from a byte array of length ${_t.length}, it MUST be 10',
```

Comment:
`10` or `at least 10`? Either check or comment is invalid.

Reply:
It must be 10 exactly. I copied from the original code and didn't pay attention. Let's fix it!